applications/CoSimulationApplication/python_scripts/convergence_accelerators/convergence_accelerator_wrapper.py

```diff
@@ -115,6 +247,17 @@ def __init__(self,
     def ComputeResidual(self, input_data):
         return self.interface_data1.GetData() - self.interface_data2.GetData()
 
+def CreateBlockResidualComputation(settings: KratosMultiphysics.Parameters,
+                                interface_data_dict: "dict[str,CouplingInterfaceData]"):
+    residual_computation_type = "data_difference"
+    if settings.Has("residual_computation"):
+        residual_computation_type = settings["residual_computation"]["type"].GetString()
+
+    if residual_computation_type == "data_difference":
+        return DataDifferenceBlockResidual(settings, interface_data_dict)
+    else:
+        raise Exception('The specified residual computation "{}" is not available!'.format(residual_computation_type))
```

Comment:
```suggestion
        raise Exception(f'The specified residual computation "{residual_computation_type}" is not available!'))
```

please always use f-strings, modern python has no place for `format` any more